### PR TITLE
Small refactor of cmd dir and fix some bugs for v1alpha2

### PR DESCRIPTION
**Pull request by fisherxu**
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/mpi-operator/109)

### CODE REVIEW

**Comment by fisherxu** [comment]
Completed part of v1alpha2 work, will add other code in follow-up PRs

/assign @rongou @terrytangyuan @gaocegege

File: pkg/controllers/v1alpha2/mpi_job_controller.go [hunk 1]
```diff
@@ -721,14 +725,21 @@ func (c *MPIJobController) updateMPIJobStatus(mpiJob *kubeflow.MPIJob, launcher
 		// 	workerStatuses.Failed += workerStatuses.Failed
 		// }
 	}
-	// Until #38113 is merged, we must use Update instead of UpdateStatus to
-	// update the Status block of the MPIJob resource. UpdateStatus will not
-	// allow changes to the Spec of the resource, which is ideal for ensuring
-	// nothing other than resource status has been updated.
-	_, err := c.kubeflowClient.KubeflowV1alpha2().MPIJobs(mpiJob.Namespace).Update(mpiJobCopy)
+
+	_, err := c.kubeflowClient.KubeflowV1alpha2().MPIJobs(mpiJob.Namespace).UpdateStatus(mpiJobCopy)
```
Comment (fisherxu) [inline pkg/controllers/v1alpha2/mpi_job_controller.go]: now we must use `UpdateStatus` to update the status.

**Comment by fisherxu** [comment]
> Have you tested examples/v1alpha2 yet?

Haven't test this tensorflow-benchmarks-imagenet, I have no environment now. The two others have tested.
Maybe we can merge this first, I will keep tested base it and send PR here.

**Comment by terrytangyuan** [comment]
/approve
Sure. Also note that there are unit tests that need to be re-enabled on Travis once they can be executed successfully.

**Comment by fisherxu** [comment]
> Also note that there are unit tests that need to be re-enabled on Travis once they can be executed successfully.

Sure, I can help fix it :)

**Pull request merged**